Anime-App/Feature/Anime/Model/QueryObject.swift

```diff
@@ -0,0 +1,16 @@
+//
+//  QueryObject.swift
+//  Anime-App
+//
+//  Created by Tobi Adegoroye on 22/08/2023.
+//
+
+import Foundation
+
+class QueryObject {
```

Comment:
This should be a struct since we don't need to observe anything in this it's just input and output, also can you change this to be `URLRequestBuilder` can you move all the logic where you create a URL in here using the URL components. This should all be done in here and this object returns a URL which you will pass into the networking manager. 

Then you can write unit tests on this.